src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/AdjournHearingsWindowMappingTest.java

```diff
@@ -0,0 +1,187 @@
+        void throwsExceptionWhenInvalidHearingDateType() {
+            SscsCaseData caseData = SscsCaseData.builder()
+                .adjournCaseNextHearingDateType("invalidType")
+                .build();
+
+            assertExceptionWhenInvalidData(caseData,
+                "Unexpected adjournCaseNextHearingDateType: invalidType");
+        }
+
+        @DisplayName("Throws exception when invalid dateOrPeriod")
+        @Test
+        void throwsExceptionWhenInvalidDateOrPeriod() {
+            SscsCaseData caseData = SscsCaseData.builder()
+                .adjournCaseNextHearingDateType(FIRST_AVAILABLE_DATE_AFTER)
+                .adjournCaseNextHearingDateOrPeriod("invalidValue")
+                .build();
+
+            assertExceptionWhenInvalidData(caseData,
+                "Unexpected adjournCaseNextHearingDateOrPeriod: invalidValue");
+        }
+
+        @DisplayName("Throws exception when missing dateOrPeriod")
+        @Test
+        void throwsExceptionWhenMissingDateOrPeriod() {
+            assertExceptionWithFirstAvailableDateAfterWhenMissingProperty(
+                null,
+                "adjournCaseNextHearingDateOrPeriod"
+            );
+        }
+
+        @DisplayName("Throws exception when missing dateAfterPeriod")
+        @Test
+        void throwsExceptionWhenMissingDateAfterPeriod() {
+            assertExceptionWithFirstAvailableDateAfterWhenMissingProperty(
+                PROVIDE_PERIOD,
+                "adjournCaseNextHearingFirstAvailableDateAfterPeriod"
+            );
+        }
+
+        @DisplayName("Throws exception when missing dateAfterDate")
+        @Test
+        void throwsExceptionWhenMissingDateAfterDate() {
+            assertExceptionWithFirstAvailableDateAfterWhenMissingProperty(
+                PROVIDE_DATE,
+                "adjournCaseNextHearingFirstAvailableDateAfterDate"
+            );
+        }
+
+        @DisplayName("Throws exception when invalid date")
+        @Test
+        void throwsExceptionWhenInvalidDate() {
+            SscsCaseData caseData = SscsCaseData.builder()
+                .adjournCaseNextHearingDateType(FIRST_AVAILABLE_DATE_AFTER)
+                .adjournCaseNextHearingDateOrPeriod(PROVIDE_DATE)
+                .adjournCaseNextHearingFirstAvailableDateAfterDate("not a date")
+                .build();
+
+            assertThatThrownBy(() -> AdjournmentCalculateDateHelper.getHearingWindowStart(caseData))
+                .isInstanceOf(DateTimeParseException.class);
+        }
+
+        private void assertExceptionWithFirstAvailableDateAfterWhenMissingProperty(
+            String dateOrPeriod,
+            String missingProperty
+        ) {
+            SscsCaseData caseData = SscsCaseData.builder()
+                .adjournCaseNextHearingDateType(FIRST_AVAILABLE_DATE_AFTER)
+                .adjournCaseNextHearingDateOrPeriod(dateOrPeriod)
+                .build();
+            // .adjournCaseNextHearingFirstAvailableDateAfterDate() and
```

Comment:
Should this be here?

Reply:
I could set them as null but that would be obsolete because they already are. I wanted to indicate that those fields are null, hence missing values.

Reply:
have set to null